core/src/main/scala/chisel3/Annotation.scala

```diff
@@ -5,18 +5,41 @@ package chisel3.experimental
 import scala.language.existentials
 import chisel3.internal.{Builder, InstanceId, LegacyModule}
 import chisel3.{CompileOptions, Data}
-import firrtl.Transform
+import firrtl.{AnnotationSeq, RenameMap, Transform}
 import firrtl.annotations._
 import firrtl.options.Unserializable
 import firrtl.transforms.{DontTouchAnnotation, NoDedupAnnotation}
 
+/** An empty annotation which does nothing
+  *
+  * Is used to implement [[ChiselAnnotation.toFirrtl]] if you instead want to return an [[AnnotationSeq]] from
+  * [[ChiselAnnotation.toAnnotationSeq]]
+  */
+case object EmptyAnnotation extends Annotation {
+  override def update(renames: RenameMap): Seq[Annotation] = Seq(this)
+}
+
 /** Interface for Annotations in Chisel
   *
   * Defines a conversion to a corresponding FIRRTL Annotation
   */
 trait ChiselAnnotation {
-  /** Conversion to FIRRTL Annotation */
+
+  /** Conversion to FIRRTL Annotation
+    * Will be deprecated in 3.5 release
+    * Please use [[toAnnotationSeq]] instead, and return [[EmptyAnnotation]] here
+    */
   def toFirrtl: Annotation
+
+  /** Conversion to FIRRTL AnnotationSeq */
+  protected def toAnnotationSeq: AnnotationSeq = Nil
+
+  private[chisel3] def convert: AnnotationSeq = {
+    toFirrtl match {
+      case EmptyAnnotation => toAnnotationSeq
+      case other => Seq(other)
+    }
+  }
```

Comment:
I'm down with the idea, just wondering about the migration path here. If we deprecate toFirrtl yet it's left as a virtual method, then every implementation of ChiselAnnotation will have a deprecation warning yet still be required to implement the method. We might consider either adding another trait and either leaving ChiselAnnotation alone or deprecating it. We could alternatively do an DependencyAPIMigration-style thing like in FIRRTL which implements the deprecated method for you.